media/server/service/source/SessionServerManager.cpp

```diff
@@ -226,6 +228,7 @@ bool SessionServerManager::switchToInactive()
 
 bool SessionServerManager::switchToNotRunning()
 {
+    std::unique_lock<std::mutex> lock{m_switchStateMutex};
```

Comment:
What's the problem here? That order of sending sendStateChangedEvent might be incorrect in extremely rare case? If I'm correct:
1. Have you managed to reproduce it without adding extra sleeps?
2. Why the mutex protects only switching to NotRunning? Other state changes can be also affected by this race, right?
3. What do you think about moving sendStateChangedEvent(common::SessionServerState::UNINITIALIZED); into ipc thread (into lambda in ApplicationManagementServer::start())? Wouldn't that synchronise the flow without mutexes?